pandas/core/tools/datetimes.py

```diff
@@ -60,7 +60,40 @@ def _maybe_cache(arg, format, cache, convert_listlike):
     return cache_array
 
 
-def _convert_and_box_cache(arg, cache_array, box, errors, name=None):
+def _box_if_needed(dt_array, box, default, tz, name=None):
```

Comment:
pls type these args with annotations

Reply:
do you mean Python type annotations?.. I don't see any Pandas code using it so far, is this something you're planning to introduce now?